File: widgetlib/src/main/java/com/zhoujiulong/widgetlib/dialog/CommonDialog.java
package com.zhoujiulong.widgetlib.dialog;

import android.app.Dialog;
import android.content.Context;
import android.support.annotation.ColorInt;
import android.support.annotation.ColorRes;
import android.support.annotation.NonNull;
import android.support.v4.content.ContextCompat;
import android.text.TextUtils;
import android.view.View;
import android.widget.TextView;
import com.zhoujiulong.widgetlib.R;

/**
 * @author zhoujiulong
 * @createtime 2019/3/7 14:03
 */
public class CommonDialog extends Dialog implements View.OnClickListener {

    private TextView mTvMsg;
    private TextView mTvLeftBt;
    private TextView mTvRightBt;
    private View mViewLine;

    private Context mContext;
    private Listener mListener;
    private boolean mIsAutoDismiss = true;

    public static CommonDialog build(Context context, String msg) {
        CommonDialog commonDialog = new CommonDialog(context);
        commonDialog.setMsg(msg);
        return commonDialog;
    }

    private CommonDialog(@NonNull Context context) {
        this(context, R.style.CommonDialog);
    }

    private CommonDialog(@NonNull Context context, int themeResId) {
        super(context, themeResId);
        setContentView(R.layout.dialog_common);
        mContext = context;

        mTvMsg = findViewById(R.id.tv_msg);
        mTvLeftBt = findViewById(R.id.tv_left);
        mTvRightBt = findViewById(R.id.tv_right);
        mViewLine = findViewById(R.id.view_center_line);

        mTvLeftBt.setOnClickListener(this);
        mTvRightBt.setOnClickListener(this);
    }

    public CommonDialog setMsg(CharSequence msg) {
        mTvMsg.setText(msg);
        return this;
    }

    public CommonDialog setLeftBtStr(String leftBtStr) {
        mTvLeftBt.setText(leftBtStr);
        if (TextUtils.isEmpty(leftBtStr)) {
            mTvLeftBt.setVisibility(View.GONE);
            mViewLine.setVisibility(View.GONE);
        } else {
            mTvLeftBt.setVisibility(View.VISIBLE);
            mViewLine.setVisibility(View.VISIBLE);
        }
        return this;
    }

    public CommonDialog setRightBtStr(String rightBtStr) {
        mTvRightBt.setText(rightBtStr);
        if (TextUtils.isEmpty(rightBtStr)) {
            mTvRightBt.setVisibility(View.GONE);
            mViewLine.setVisibility(View.GONE);
        } else {
            mTvRightBt.setVisibility(View.VISIBLE);
            mViewLine.setVisibility(View.VISIBLE);
        }
        return this;
    }

    public CommonDialog setListener(Listener listener) {
        mListener = listener;
        return this;
    }

    public CommonDialog setMsgTextColorRes(@ColorRes int colorRes) {
        return setMsgTextColor(ContextCompat.getColor(getContext(), colorRes));
    }

    public CommonDialog setMsgTextColor(@ColorInt int colorRes) {
        mTvMsg.setTextColor(colorRes);
        return this;
    }

    public CommonDialog setLeftBtTextColorRes(@ColorRes int colorRes) {
        return setLeftBtTextColor(ContextCompat.getColor(getContext(), colorRes));
    }

    public CommonDialog setLeftBtTextColor(@ColorInt int colorRes) {
        mTvLeftBt.setTextColor(colorRes);
        return this;
    }

    public CommonDialog setRightBtTextColorRes(@ColorRes int colorRes) {
        return setRightBtTextColor(ContextCompat.getColor(getContext(), colorRes));
    }

    public CommonDialog setRightBtTextColor(@ColorInt int colorRes) {
        mTvRightBt.setTextColor(colorRes);
        return this;
    }

    public CommonDialog setAutoDimiss(boolean autoDimiss) {
        mIsAutoDismiss = autoDimiss;
        return this;
    }

    @Override
    public void onClick(View view) {
        if (mIsAutoDismiss) dismiss();
        int id = view.getId();
        if (id == R.id.tv_left) {
            if (mListener != null) mListener.onLeftClick();
        } else if (id == R.id.tv_right) {
            if (mListener != null) mListener.onRightClick();
        }
    }

    public static abstract class Listener {
        public abstract void onRightClick();

        public void onLeftClick() {
        }
    }

}

File: widgetlib/src/main/java/com/zhoujiulong/widgetlib/dialog/CommonDialog.kt
package com.zhoujiulong.widgetlib.dialog

import android.app.Dialog
import android.content.Context
import android.support.annotation.ColorInt
import android.support.annotation.ColorRes
import android.support.v4.content.ContextCompat
import android.text.TextUtils
import android.view.View
import android.widget.TextView
import com.zhoujiulong.widgetlib.R

/**
 * @author zhoujiulong
 * @createtime 2019/3/7 14:03
 */
class CommonDialog private constructor(
    mContext: Context,
    themeResId: Int = R.style.CommonDialog
) : Dialog(mContext, themeResId), View.OnClickListener {

    private val mTvMsg: TextView
    private val mTvLeftBt: TextView
    private val mTvRightBt: TextView
    private val mViewLine: View
    private var mListener: Listener? = null
    private var mIsAutoDismiss = true

    init {
        setContentView(R.layout.dialog_common)

        mTvMsg = findViewById(R.id.tv_msg)
        mTvLeftBt = findViewById(R.id.tv_left)
        mTvRightBt = findViewById(R.id.tv_right)
        mViewLine = findViewById(R.id.view_center_line)

        mTvLeftBt.setOnClickListener(this)
        mTvRightBt.setOnClickListener(this)
    }

    fun setMsg(msg: CharSequence): CommonDialog {
        mTvMsg.text = msg
        return this
    }

    fun setLeftBtStr(leftBtStr: String): CommonDialog {
        mTvLeftBt.text = leftBtStr
        if (TextUtils.isEmpty(leftBtStr)) {
            mTvLeftBt.visibility = View.GONE
            mViewLine.visibility = View.GONE
        } else {
            mTvLeftBt.visibility = View.VISIBLE
            mViewLine.visibility = View.VISIBLE
        }
        return this
    }

    fun setRightBtStr(rightBtStr: String): CommonDialog {
        mTvRightBt.text = rightBtStr
        if (TextUtils.isEmpty(rightBtStr)) {
            mTvRightBt.visibility = View.GONE
            mViewLine.visibility = View.GONE
        } else {
            mTvRightBt.visibility = View.VISIBLE
            mViewLine.visibility = View.VISIBLE
        }
        return this
    }

    fun setListener(listener: Listener): CommonDialog {
        mListener = listener
        return this
    }

    fun setMsgTextColorRes(@ColorRes colorRes: Int): CommonDialog {
        return setMsgTextColor(ContextCompat.getColor(context, colorRes))
    }

    fun setMsgTextColor(@ColorInt colorRes: Int): CommonDialog {
        mTvMsg.setTextColor(colorRes)
        return this
    }

    fun setLeftBtTextColorRes(@ColorRes colorRes: Int): CommonDialog {
        return setLeftBtTextColor(ContextCompat.getColor(context, colorRes))
    }

    fun setLeftBtTextColor(@ColorInt colorRes: Int): CommonDialog {
        mTvLeftBt.setTextColor(colorRes)
        return this
    }

    fun setRightBtTextColorRes(@ColorRes colorRes: Int): CommonDialog {
        return setRightBtTextColor(ContextCompat.getColor(context, colorRes))
    }

    fun setRightBtTextColor(@ColorInt colorRes: Int): CommonDialog {
        mTvRightBt.setTextColor(colorRes)
        return this
    }

    fun setAutoDimiss(autoDimiss: Boolean): CommonDialog {
        mIsAutoDismiss = autoDimiss
        return this
    }

    override fun onClick(view: View) {
        if (mIsAutoDismiss) dismiss()
        val id = view.id
        if (id == R.id.tv_left) {
            mListener?.onLeftClick()
        } else if (id == R.id.tv_right) {
            mListener?.onRightClick()
        }
    }

    abstract class Listener {
        abstract fun onRightClick()

        fun onLeftClick() {}
    }

    companion object {

        fun build(context: Context, msg: String): CommonDialog {
            val commonDialog = CommonDialog(context)
            commonDialog.setMsg(msg)
            return commonDialog
        }
    }

}















